mozilla-services/socorro
# fix bug 1346891: rewrite BetaVersionRule to use webapp API

This:

1. creates a webapp API called VersionString that returns ... well, ... version strings
2. creates a new `ExpiringCache` that has a ttl in addition to max size because we didn't have one and I wasn't enamored with the ones I saw (though `cachetools` might be a good choice)
3. rewrites the `BetaVersionRule` to use the new cache and new VersionString API
4. implements a `session_with_retries` and rewrites everything to use that rather than using requests directly

So... there's a lot of different things in this PR all lumped together, but it's all sort of related.

To test:

1. go to `http://localhost:8000/api/` and test the VersionString API using product/version/builds
2. set `processor.version_string_api` to `http://localhost:8000/api/VersionString` in your `my.env` file, then process a crash from the beta channel with a build that's in your db and make sure it gets the right version
